README.md

```diff
@@ -197,3 +197,126 @@ e3db?.revoke(type: "UserInfo", readerId: otherClient) { result in
+
+The E3DB SDK allows you to encrypt documents for local storage, which can
+be decrypted later, by the client that created the document or any client with
+which the document has been `shared`. Note that locally encrypted documents
+_cannot_ be written directly to E3DB -- they must be decrypted locally and
+written using the `write` or `update` methods.
+
+Local encryption (and decryption) requires two steps:
+
+1. Create a 'writer key' (for encryption) or obtain a 'reader key' (for
+  decryption).
+2. Call `encrypt` to encrypt a new document for decryption, call `decrypt`.
+
+The 'writer key' and 'reader key' are both `EAKInfo` objects. An `EAKInfo`
+object holds an encrypted key that can be used by the intended client to encrypt
+or decrypt associated documents. A writer key can be created by calling
+`createWriterKey`; a 'reader key' can be obtained by calling `getReaderKey`.
+(Note that the client calling `getReaderKey` will only receive a key if the
+writer of those records has given access to the calling client through the
+`share` operation.)
+
+The `createWriterKey` and `getReaderKey` are networked operations, (which means
+they are asynchronous operations as well), but can be performed once ahead of
+time. The `EAKInfo` instances returned from those operations are safe to store
+locally, and can be used in the non-networked operations of `encrypt` and
+`decrypt`.
+
+Here is an example of encrypting a document locally:
+
+```swift
+let recordData = RecordData(cleartext: ["SSN": "123-45-6789"])
+let recordType = "UserInfo"
+
+e3db?.createWriterKey(type: type) { result in
+    switch result {
+    // The operation was successful, here's the `EAKInfo` instance,
+    // you can think of this as the "encryption key", but it's also encrypted,
+    // so you don't have to worry about storing it in plaintext or exposing it.
+    case .success(let eak):
+        // attempt to create an encrypted document with the EAKInfo
+        let encrypted = try? client1!.encrypt(type: recordType, data: recordData, eakInfo: eak)
+        print("Encrypted document: \(encrypted!)")
+
+    case .failure(let error):
+        print("An error occurred attempting to create writer key: \(error)")
+    }
+}
+```
+
+(Note that the `EAKInfo` instance is safe to store with the encrypted data, as
+it is also encrypted). The client can decrypt the given record as follows:
+
+```swift
+let encrypted = // get encrypted document (e.g. read from local storage)
+let writerKey = // get stored EAKInfo instance (e.g. from local storage)
+
+let decrypted = try e3db?.decrypt(encryptedDoc: encrypted, eakInfo: writerKey)
+```
+
+##### Local Decryption of Shared Records
+
+When two clients have a sharing relationship, the "reader" can locally decrypt
+any documents encrypted by the "writer," without using E3DB for storage.
+
+The 'writer' must first share records with a 'reader', using the `share` method.
+The 'reader' can then decrypt any locally encrypted records as follows:
```

Comment:
Add a note describing how to get the reader key, something like:

> ... `share` method. The reader must then obtain a reader using `getReaderKey`. Note that this is a networked operation. However, the `EAKInfo` object can be saved for later use. After obtaing a reader key, the `reader` then then decrypt any records encrypted by the writer as follows:

Also I emphasized that the reader is decrypting documents encrypted by the writer, not "any locally encrypted" record.

Reply:
Also I just realized this text comes from the Android SDK so I'll port any changes you make here back to that README. :)